hibernate-core/src/main/java/org/hibernate/engine/internal/Cascade.java

```diff
@@ -96,6 +96,12 @@ public static void cascade(
 					if ( hasUninitializedLazyProperties && persister.getPropertyLaziness()[i] && action.performOnLazyProperty() ) {
 						LazyAttributeLoadingInterceptor interceptor = persister.getInstrumentationMetadata().extractInterceptor( parent );
 						child = interceptor.fetchAttribute( parent, propertyName );
```

Comment:
Explicitly set the fetched attribute in the parent.

Should `LazyAttributeLoadingInterceptor#fetchAttribute` set the attribute in the parent instead?

If the attribute is not set in the parent, then during flush, it will appear that the fetched collection is unreferenced.